app/src/main/java/com/findmyrecycling/service/ProductService.kt

```diff
@@ -8,13 +9,17 @@ import kotlinx.coroutines.async
 import kotlinx.coroutines.withContext
 import retrofit2.awaitResponse
 
-class ProductService{
 
-    suspend fun fetchProducts() : List<Product>? {
+interface IProductService {
+    suspend fun fetchProducts() : List<Product>?
+}
+
+class ProductService : IProductService {
+    override suspend fun fetchProducts() : List <Product>? {
         return withContext(Dispatchers.IO) {
             val service = RetrofitClientInstance.retrofitInstance?.create(IProductDAO::class.java)
-            val products = async {service?.getAllProducts()}
-            var result = products.await()?.awaitResponse()?.body()
+            val items = async { service?.getAllProducts()}
+            var result = items.await()?.awaitResponse()?.body()
             return@withContext result
         }
```

Comment:
Implemented interface to remove concretion.
https://oozou.com/blog/interface-in-kotlin-and-when-to-use-it-10